docs/source/entities-advanced.mdx

```diff
@@ -6,13 +6,39 @@ This article describes complex behaviors of federated entities beyond those cove
 
 ## Advanced `@key`s
 
-A single entity can have multiple `@key`s. Additionally, a `@key` can include multiple fields, and even arbitrarily nested fields.
+A single `@key` can require a combination of fields to uniquely identify an entity. Additionally, an entity can have multiple separate `@key`s if multiple fields or combinations of fields can independently serve as unique identifiers.
+
+### Compound `@key`s
+
+A single `@key` can consist of multiple fields, the combination of which is required to uniquely identify an entity. This is called a **compound** or composite key. In the following example, the combination of both `username` and `domain` fields is required to uniquely identify the `User` entity:
+
+```graphql {1} title="Users subgraph"
+type User @key(fields: "username domain") {
+  username: String!
+  domain: String!
+}
```

Comment:
Add, or link to, explanation: must a @key field be required (`!`)? What happens if a @key field isn't required?

Reply:
Good callout! I'd assume keys can only use non-nullable field but based on [this conversation](https://apollograph.slack.com/archives/C03RH11Q6S1/p1692821163122989), it seems that's not the case.

I think this info is better suited for the "basics" page and will add there. https://github.com/apollographql/federation/pull/2771/commits/df834d31809f996a8c27331545fe08e5c92cbfd1

Reply:
yep ideally `@key`s are non-nullable but they can be nullable